File: JanperCab/3-Application/Dtos/Enquiry/EnquiryPriceDto.cs
namespace _3_Application.Dtos.Enquiry
{
    public abstract class EnquiryPriceDto
    {
        public int Id { get; set; }

        public bool ToBePriced { get; set; }

        public bool IsShippingRequired { get; set; }

        public bool HasFixedPrice { get; set; }

        public decimal DeliveryFee { get; set; }

        public decimal SubTotal { get; set; }

        public decimal TotalGst { get; set; }

        public decimal? TotalPrice { get; set; }
    }
}

File: JanperCab/5-JanperCab/ClientApp/src/app/_models/enquiry/EnquiryPriceDto.ts
import { EnquiryDto } from './EnquiryDto';

export abstract class EnquiryPriceDto {
  $type: string;
  id: number;

  toBePriced: boolean;
  isShippingRequired: boolean;
  hasFixedPrice: boolean;
  deliveryFee: number;
  subTotal: number;
  totalGst: number;
  totalPrice: number;

  constructor(enquiry: EnquiryDto) {
    this.$type = '_3_Application.Dtos.Enquiry.EnquiryDto, 3-Application';
    this.id = enquiry.id;
    this.toBePriced = enquiry.toBePriced;
    this.isShippingRequired = enquiry.isShippingRequired;
    this.hasFixedPrice = enquiry.hasFixedPrice;
    this.deliveryFee = enquiry.deliveryFee;
    this.subTotal = enquiry.subTotal;
    this.totalGst = enquiry.totalGst;
    this.totalPrice = enquiry.totalPrice;
  }
}
